vllm/worker/model_runner.py

```diff
@@ -1668,6 +1702,45 @@ def execute_model(
 
         return [output]
 
+    def recv_kv_needed(self, model_input, kv_caches) -> bool:
+        """
+            Need to receive KV when
+            1. current vLLM instance is KV cache *consumer*
+            2. this batch is not a profiling run
+            3. this batch is a prefill run
+        """
+
+        prefill_meta = model_input.attn_metadata.prefill_metadata
+
+        # check if the current run is profiling
+        is_profile_run = (kv_caches is None) or (kv_caches[0] is None)
+        # check if the current run is prefill
+        is_prefill_run = prefill_meta is not None
+
+        return all([
+            dist_kv.IS_KV_CONSUMER,
+            not is_profile_run,
+            is_prefill_run,
+        ])
```

Comment:
Not sure if `and` would be faster than `all` (or it's opposite)?

Reply:
Yeah and is faster. I switched to and.